src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -44,6 +44,19 @@ def event_activate(self):
             f'{self.channel_prefix}/device_info': self._device_info,
         })
 
+    def _is_ess_buy_limit_surpassed(self, energy):
+        return energy > self.state.energy_to_buy_dict[self.market.time_slot]
+
+    def _is_ess_sell_limit_surpassed(self, energy):
+        return energy > self.state.energy_to_sell_dict[self.market.time_slot]
+
+    def _publish_error_message_to_response_channel(self, response_channel, transaction_id):
```

Comment:
I would strongly suggest to remove this function and to revert the assert method. Reason for that, this function is used for both bids and offers, but the command field of the response is always 'bid'. Moreover, the status: 'error' field is not reported. Both these issues will be fixed if you revert to the former implementation.